### PR TITLE
app: fix zeros stil being ignored, and clamping out-of-bounds values instead of ignoring them

### PR DESCRIPTION
**Pull request by ScottSWu**
nxplayer bin_2023-01-19-11-18-25-148

https://app.asana.com/0/1203288718560208/1203665494438008/f

### CODE REVIEW

File: app/packages/looker/src/overlays/heatmap.ts [hunk 1]
```diff
@@ -199,14 +199,18 @@ export default class HeatmapOverlay<State extends BaseState>
   private getColor(state: Readonly<State>, value: number): number {
```
Comment (ScottSWu) [inline app/packages/looker/src/overlays/heatmap.ts:199]: I'm not quite sure where this version of `getColor` is used, but I added clamp here too anyways.

File: app/packages/looker/src/overlays/segmentation.ts [hunk 2]
```diff
@@ -78,7 +78,7 @@ export default class SegmentationOverlay<State extends BaseState>
       pixelCoordinates: [x, y],
       dimensions: [w, h],
     } = state;
-    if (x >= 0 && x <= w && y >= 0 && y <= h && this.getTarget(state)) {
+    if (x >= 0 && x <= w && y >= 0 && y <= h) {
```
Comment (ScottSWu) [inline app/packages/looker/src/overlays/segmentation.ts:81]: Also preventing the tooltips from showing, but for segmentations.

File: app/packages/looker/src/overlays/heatmap.ts [hunk 3]
```diff
@@ -93,7 +93,7 @@ export default class HeatmapOverlay<State extends BaseState>
       pixelCoordinates: [x, y],
       dimensions: [w, h],
     } = state;
-    if (x >= 0 && x <= w && y >= 0 && y <= h && this.getTarget(state)) {
+    if (x >= 0 && x <= w && y >= 0 && y <= h) {
```
Comment (ScottSWu) [inline app/packages/looker/src/overlays/heatmap.ts:96]: This was preventing the tooltip from showing when hovering over a 0 value.

File: app/packages/looker/src/worker.ts [hunk 4]
```diff
@@ -458,9 +456,7 @@ const UPDATE_LABEL = {
 
     // these for loops must be fast. no "in" or "of" syntax
     for (let i = 0; i < overlay.length; i++) {
-      if (targets[i] !== 0) {
```
Comment (ScottSWu) [inline app/packages/looker/src/worker.ts:461]: Here's where it was doing another filtering of 0s.

File: app/packages/looker/src/worker.ts [hunk 5]
```diff
@@ -498,9 +494,7 @@ const UPDATE_LABEL = {
...
```
Comment (ScottSWu) [inline app/packages/looker/src/worker.ts:501]: Same filtering of 0s but for segmentation.

**Review by konstantin-azarov — APPROVED** [review]

**Pull request merged**